react-components/src/components/Widgets/SplitWidget.tsx

```diff
@@ -0,0 +1,133 @@
+/*!
+ * Copyright 2023 Cognite AS
+ */
+
+import { Button } from '@cognite/cogs.js';
+import { type ReactElement, useState, type ReactNode, useEffect } from 'react';
+import styled from 'styled-components';
+import Widget from './Widget';
+import Draggable, { type DraggableData, type DraggableEvent } from 'react-draggable';
+import { useRevealContainerElement } from '../RevealContainer/RevealContainerElementContext';
+import { ResizableBox } from 'react-resizable';
+import 'react-resizable/css/styles.css';
+import { HEIGHT_FACTOR, WIDTH_FACTOR, X_OFFSET, Y_OFFSET } from './constants';
+
+type SplitWidgetProps = {
+  title?: string;
+  subtitle?: string;
+  header?: string;
+  type?: string;
+  children: ReactNode;
+};
+
+export const SplitWidget = ({
+  title,
+  subtitle,
+  header,
+  type,
+  children
+}: SplitWidgetProps): ReactElement => {
+  const [isMinimized, setIsMinimized] = useState(false);
+  const [position, setPosition] = useState({ x: 0, y: 0 });
+  const [size, setSize] = useState({ width: 0, height: 0 });
+  const parentContainerElement = useRevealContainerElement();
+
+  if (parentContainerElement === undefined) {
+    return <></>;
+  }
+
+  useEffect(() => {
+    const updateSize = (): void => {
+      const parentWidth = parentContainerElement.clientWidth;
+      const parentHeight = parentContainerElement.clientHeight;
+
+      const width = isMinimized ? 300 : parentWidth * WIDTH_FACTOR;
+      const height = isMinimized ? 48 : parentHeight * HEIGHT_FACTOR;
+
+      setSize({ width, height });
+    };
+
+    updateSize();
+
+    window.addEventListener('resize', updateSize);
+
+    return () => {
+      window.removeEventListener('resize', updateSize);
+    };
+  }, [isMinimized]);
+
+  const handleExpand = (): void => {
+    setIsMinimized((prev) => !prev);
+  };
+
+  const handleDrag = (event: DraggableEvent, data: DraggableData): void => {
+    event.stopPropagation();
+
+    const { x, y, node } = data;
+    const { left, top, width } = node.getBoundingClientRect();
+    const {
+      width: parentWidth,
+      height: parentHeight,
+      left: parentLeft,
+      top: parentTop
+    } = parentContainerElement.getBoundingClientRect();
+
+    if (
+      left + width - X_OFFSET < parentLeft ||
+      top < parentTop ||
+      left + X_OFFSET > parentWidth ||
+      top + Y_OFFSET > parentHeight
+    ) {
+      // Prevent moving beyond the canvas
+      return;
+    }
```

Comment:
Is this working as expected? I can move the window off-canvas, but it moves back to the canvas when I let go of my mouse button. Is that intended?

Reply:
Also.. Perhaps it would be nice to only move it as much as is needed? So, not return it to its original position, but clamp it to the edge?

Reply:
> Is this working as expected? I can move the window off-canvas, but it moves back to the canvas when I let go of my mouse button. Is that intended?

Yes, I had check this scenario with Kerem and had made the window restricted to be within the parent window even if the window is moved out of parent